### PR TITLE
Ako/feq 701/setup GitHub workflows

### PR DESCRIPTION
**Pull request by ako-deriv**
# Changes
* Added Slack action
* Added Publish to Cloudflare Staging and Production 
* Added Slack notification Action
* Added Production Workflow
* Added Staging Workflow

### CODE REVIEW

File: .github/workflows/release-staging.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      # setup nodejs environment
+      - name: Setup Node.js environment
+        uses: ./.github/actions/setup_node
+      # cache the dependencies to speed up the build and then install dependencies
+      - name: Cache dependencies
+        uses: ./.github/actions/npm_install_from_cache
+      # build the react app
+      - name: Build
+        run: npm run build
+      # run tests
```
Comment (niloofar-deriv) [inline .github/workflows/release-staging.yml]: We need to save our cache 🤔 where we should handle that?

Comment (niloofar-deriv) [inline .github/workflows/release-staging.yml]: You commented run test, but we don't have any script here for running our tests 🤔 
Should it be sth like this?
```suggestion
      # Publish to Cloudflare Staging
```

File: .github/workflows/release-staging.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        uses: actions/checkout@v4
+      # setup nodejs environment
+      - name: Setup Node.js environment
+        uses: ./.github/actions/setup_node
+      # cache the dependencies to speed up the build and then install dependencies
+      - name: Cache dependencies
+        uses: ./.github/actions/npm_install_from_cache
+      # build the react app
+      - name: Build
+        run: npm run build
```
Comment (niloofar-deriv) [inline .github/workflows/release-staging.yml:21]: I think we need to run our typescript too 🤔

Comment (niloofar-deriv) [reply .github/workflows/release-staging.yml:21]: And tests 🤔

Comment (ako-deriv) [reply .github/workflows/release-staging.yml:21]: Test is done.
for tsc we need to come up with the command and configs first

Comment (niloofar-deriv) [reply .github/workflows/release-staging.yml:21]: Can we do it in this PR or later we need to create another card for it?

Comment (ako-deriv) [reply .github/workflows/release-staging.yml:21]: the scope is different lets test it on another card.

**Review by niloofar-deriv — APPROVED** [review]

File: .github/actions/publish_to_pages_production/action.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  using: composite
+  steps:
+  - name: Publish to cloudflare pages (production)
+    env:
+      CLOUDFLARE_ACCOUNT_ID: ${{ inputs.CLOUDFLARE_ACCOUNT_ID }}
+      CLOUDFLARE_API_TOKEN: ${{ inputs.CLOUDFLARE_API_TOKEN }}
+    run: |
+      npm i wrangler@3.1.0
+      npx wrangler pages deploy dist/ --project-name=deriv-smarttrader --branch=main
+      echo "New website - http://dsmarttrader.deriv.com"
```
Comment (matin-deriv) [inline .github/actions/publish_to_pages_production/action.yml]: Nit
```suggestion
      echo "New website - https://dsmarttrader.deriv.com"
```

File: .github/actions/publish_to_pages_staging/action.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  using: composite
+  steps:
+  - name: Publish to cloudflare pages (staging)
+    env:
+      CLOUDFLARE_ACCOUNT_ID: ${{ inputs.CLOUDFLARE_ACCOUNT_ID }}
+      CLOUDFLARE_API_TOKEN: ${{ inputs.CLOUDFLARE_API_TOKEN }}
+    run: |
+      npm i wrangler@3.1.0
+      npx wrangler pages deploy dist/ --project-name=deriv-smarttrader --branch=staging
+      echo "New staging website - http://staging-dsmarttrader.deriv.com"
```
Comment (matin-deriv) [inline .github/actions/publish_to_pages_staging/action.yml]: ditto.

**Review by matin-deriv — DISMISSED** [review]

**Review by niloofar-deriv — APPROVED** [review]

**Review by matin-deriv — APPROVED** [review]

**Pull request merged**